src/promptflow-tools/promptflow/tools/common.py

```diff
@@ -84,7 +84,7 @@ def validate_functions(functions):
 def parse_function_role_prompt(function_str):
     # customer can add ## in front of name/content for markdown highlight.
     # and we still support name/content without ## prefix for backward compatibility.
-    pattern = r"\n*#{0,2}\s*name:\n\s*(\S+)\s*\n*#{0,2}\s*content:\n(.*)"
+    pattern = r"\n*#{0,2}\s*name:\n+\s*(\S*)\s*\n*#{0,2}\s*content:\n?(.*)"
```

Comment:
please explain about the change.

Reply:
new change `\n+` means allow more than one `\n` after `name:`. Update `\S+` to `\S*` because we allow there is no content after name. `content:\n?(.*)` matches two conditions: original pattern `content:\n(.*)` or `content:` which allows no content after `content`.

Reply:
revert \S* to \S+